### ISSUE TITLE
Add more node sounds

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#6355]
Sounds in Minetest are still a bit dull overall. I suggest to add a couple of node sounds to give subgame a chance to improve.

Ideas for new node sounds:

* [ ] Falling node hits the ground
* [ ] Falling node starts to fall
* [ ] Attached node detaches
* [ ] Liquid physically flows by one step. This sound is played once per “flow step”, it is NOT looped! It might be required to reduce the amount of sounds played, otherwise it could become loud quickly
* [ ] Looped sound played to player while being inside with the head (good for underwater “bubbling” sound)
* For objects which do footstep sounds (players, mobs):
    * [ ] Fall on a node. If sound is not set, fall back to `footstep`
    * [ ] Jump off a node (or fail trying if `disable_jump=1`)
    * [ ] Bounce off a bouncy node from any direction. Maybe make this sound louder the higher the bounce
    * [ ] Pass through a node with any part of the body (useful for non-`walkable` plants)
        * [ ] Maybe: Also replace the odd overloading of footstep sound for liquids with this
    * [ ] Enter this type of node type from any direction with any part of the body at a high speed. Only played once after entry. To play it again, the object has to be completely submerged in any different node type, then the sound can be triggered again (useful for loud water splash effects)

Feel free to comment my ideas or add your own.

**Comment by paramat** [comment]
> Footstep nodes - Fall on a node

> Jump off a node 

I feel the footstep sound is already close enough to be used for these, although maybe the same sound could be played a little louder.

**Comment by DS-Minetest** [comment]
Related: #2272

**Comment by C1ffisme** [comment]
This isn't really a specific sound in general, but it might be interesting if we could muffle sounds if the player is in certain blocks like liquids.

It doesn't sound too complicated, the server can send a message to the client that they are in a muffling block, then the client will put an effect on all the sounds played to the player.

**Comment by Wuzzy2** [comment]
> the server can send a message to the client that they are in a muffling block

Bad idea as this requires an avoidable round trip and a delay. Doesn't the client already know its position? I believe the client is already aware of node definitions as well. So the client really just needs to infer the behaviour from what it already knows, no need to bother the server again.
Either add this setting to the node definition or define “muffling block” implicitly as any block which is a liquid or walkable.

Oh, wait. This is off-topic.

**Comment by Ferk** [comment]
For the falling sound, maybe play 2 separate footstep sounds in succession but almost immediately one after the other without too much of a pause.
That's normally how falling on your feet sounds.

**Comment by Ferk** [comment]
Yes, I expect all of these to be triggered client-side, just like footsteps are client side. The server just needs to provide the sound files and the client will play them in the appropriate times, coded in the engine.

**Comment by paramat** [comment]
I support some of these or at least considering some of these.

**Comment by paramat** [comment]
Sorry i should have mentioned which i supported and which i didn't.

I disagree with (reasons in #7719):
Falling node starts to fall
Attached node detaches

Maybe:
 Jump off a node
 Bounce off a bouncy node from any direction

Liquid physically flows by one step: Not needed if we have ambience sounds for a flowing liquid, which is preferable.

Fall on a node: Already happens.

**Comment by Wuzzy2** [comment]
I should first clarify a very important principle of all my sound suggestions:
In general, any sound additions we make to nodes should always have a defined fallback behaviour if the sound is not defined (`nil`), so we do not actually force mods to use any particular sounds if they don't want them.
The usual fallback behaviour for most sounds will be just “play no sound”. But sometimes a closely-related sound is desirable as fallback: E.g. if you fall on a node, but the “fall on node” sound is missing, play “footstep” instead.

So, in general, I want to have more options to add sounds for various events, but always with fallback options. This keeps complexity for sound makers low while still adding flexibility for those who want it.

@paramat:

> Falling node starts to fall
> Attached node detaches

I disputed both rejections in #7719.

> Liquid physically flows by one step: Not needed if we have ambience sounds for a flowing liquid, which is preferable.

Agreed, this is a good point. Real ambient sounds for liquids would be a very good improvement indeed.

> Fall on a node: Already happens.

Yes, but it's only the footstep sound.
A different sound is not possible. This can be optional, but it would add more variety if more sounds are possible. Falling on a node and walking on a node are two different things and should (if desired by the game) sound different. Of course, the fallback here would be “footstep” (as we already do).

> Jump off a node (use footstep sound)

Kinda, but it would be better (more artistic freedom) if a separate sound is allowed, with footstep as the fallback. Note it would not significantly increase the complexity either.

**Comment by paramat** [comment]
For falling on a node, a louder footstep sound would be so ideal there's no point in addng another sound. It's the sound of feet hitting a surface, the only difference betweem walking and falling-onto is the velocity of a hit, so more volume is all you need. It would be very difficult to think of a necessary sound other than a footstep type sound. A separate sound would be a silly level of complexity.

Jumping off a surface could be of use for a more abstract game where you want a cartoonish BWOO! or HUP! sound for a jump. Also realistically, it is reasonable that it would be a different type of sound, so yeah maybe, i'll edit my post.

**Comment by DS-Minetest** [comment]
>the only difference betweem walking and falling-onto is the velocity of a hit, so more volume is all you need.

I think, the speed and length of the sound can be different, too.

**Comment by sfan5** [comment]
@Wuzzy2 how much of this is done?

**Comment by Wuzzy2** [comment]
* [x] Falling node starts to fall (`sounds.fall`)
* [x] Attached node detaches (`sounds.fall`)
* [ ] Pass through a node with any part of the body (useful for non-walkable plants, vines, cobwebs): partially implemented. `footstep` sound is played when moving through liquids. But there's no sound for anything else)
* [ ] Everything else
